### PR TITLE
WIP: Working on HTTP2 client

### PR DESCRIPTION
**Pull request by omarkj**
This is an ongoing PR to work on the HTTP2 client that comes with Chatterbox.

My goals are to:
* Clean up the API (it's pretty nice now, tho);
* Gracefully handle failure
* Somehow make it easy to work with multiple streams over a single connection.

For now I'm not looking to:
* Support HTTP/1.1 downgrading

### CODE REVIEW

File: src/http2_connection.erl [hunk 1]
```diff
@@ -601,7 +601,7 @@ handle_info({inet_async, _ListSock, Ref, {ok, CliSocket}},
         ssl ->
             {ok, AcceptSocket} = ssl:ssl_accept(CliSocket, SSLOptions),
             %% TODO: Erlang 18 uses ALPN
-            {ok, _Upgrayedd} = ssl:negotiated_next_protocol(AcceptSocket),
```
Comment (joedevivo) [inline src/http2_connection.erl]: IIRC this needs to be a switch based on erlang version 17/18

Comment (omarkj) [inline src/http2_connection.erl]: You are right. What is a sane way of doing that? Macro party?

Comment (joedevivo) [inline src/http2_connection.erl]: probably

**Comment by joedevivo** [comment]
Hey, take a look at what I've done in http2_client on master.

**Comment by joedevivo** [comment]
a bunch of client dev has happened since this was opened.

**Pull request closed without merging**